larray/core/array.py

```diff
@@ -6023,15 +6028,15 @@ def to_excel(self, filepath=None, sheet_name=None, position='A1', overwrite_file
             wb = open_excel(filepath, overwrite_file=overwrite_file)
 
             if new_workbook:
-                sheet = wb.sheets[0]
-                if sheet_name is not None:
-                    sheet.name = sheet_name
-            elif sheet_name is not None and sheet_name in wb:
-                sheet = wb.sheets[sheet_name]
+                first_sheet = wb.sheets[0]
+                if sheet is not None:
```

Comment:
Maybe our tests are not good enough to catch this, but this is buggy... What you need to do is rename all occurrences of "sheet" (before the sheet_name -> sheet rename, obviously) to something else. E.g. sheetobj